src/writer/serve.py

```diff
@@ -601,13 +676,43 @@ def _mount_server_static_path(app: FastAPI, server_static_path: pathlib.Path) ->
 
     Writer Framework routes remain priority. A developer cannot come and overload them.
     """
-    app.get('/')(lambda: FileResponse(server_static_path.joinpath('index.html')))
     for f in wf_root_static_assets():
         if f.is_file():
             app.get(f"/{f.name}")(lambda: FileResponse(f))
         if f.is_dir():
             app.mount(f"/{f.name}", StaticFiles(directory=f), name=f"server_static_{f}")
 
+def _mount_render_index_html(app: FastAPI, server_static_path: pathlib.Path):
+    """
+    Serves the main page with the title that has been configured.
+
+    :param app:
+    :param server_static_path:
+    :return:
+    """
+    def _render_index_html():
+        with io.open(server_static_path.joinpath('index.html'), 'r', encoding='utf-8') as f:
+            index_html = f.read()
+            if hasattr(app.state, "title"):
+                index_html = index_html.replace("<title>Writer Framework</title>", f"<title>{app.state.title}</title>")
+
+            if hasattr(app.state, "meta"):
+                meta = app.state.meta() if callable(app.state.meta) else app.state.meta
+                meta_tags = "\n".join([f'<meta name="{k}" content="{v}">' for k, v in meta.items()])
+                index_html = index_html.replace("<!-- {{ meta }} -->", meta_tags)
+            else:
+                index_html = index_html.replace("<!-- {{ meta }} -->", "")
+
+            if hasattr(app.state, "opengraph_tags"):
+                opengraph_tags = app.state.opengraph_tags() if callable(app.state.opengraph_tags) else app.state.opengraph_tags
+                opengraph_tags = "\n".join([f'<meta property="{k}" content="{v}">' for k, v in opengraph_tags.items()])
```

Comment:
The injected variables can contain special characters or even text from the DB as [your example demonstrate it](https://github.com/writer/writer-framework/blob/b9ba3e8b079523be70cfc03f11bc8175cf70d325/docs/framework/seo.mdx?plain=1#L74-L79).

So it can lead to:

- broken HTML if the variable include special char like `"`
- or even XSS attack if someone succeed to insert `<script>` in DB (XSS attack)

You must sanitize the variables using [html.escape](https://docs.python.org/3/library/html.html#html.escape)

```suggestion
                index_html = index_html.replace("<title>Writer Framework</title>", f"<title>{app.state.title}</title>")

            if hasattr(app.state, "meta"):
                meta = app.state.meta() if callable(app.state.meta) else app.state.meta
                meta_tags = "\n".join([f'<meta name="{escape(k)}" content="{escape(v)}">' for k, v in meta.items()])
                index_html = index_html.replace("<!-- {{ meta }} -->", meta_tags)
            else:
                index_html = index_html.replace("<!-- {{ meta }} -->", "")

            if hasattr(app.state, "opengraph_tags"):
                opengraph_tags = app.state.opengraph_tags() if callable(app.state.opengraph_tags) else app.state.opengraph_tags
                opengraph_tags = "\n".join([f'<meta property="{escape(k)}" content="{escape(v)}">' for k, v in opengraph_tags.items()])
```

Reply:
Ok for the `content` that may come from a database. I would avoid for the name it's supposed to be checked by the developper.